### PR TITLE
Romanchuk lukashov/chat provider admin add

### PR DESCRIPTION
**Pull request by Sandaljero**
Functionality for provider's deputy and provider's admin was added

### CODE REVIEW

File: OutOfSchool/OutOfSchool.DataAccess/Repository/ChatRoomWorkshopModelForChatListRepository.cs [hunk 1]
```diff
@@ -61,6 +61,16 @@ public Task<List<ChatRoomWorkshopForChatList>> GetByWorkshopIdAsync(Guid worksho
             return this.GetByParametersAsync(condition, searchMessagesForProvider);
         }
 
+        /// <inheritdoc/>
+        public Task<List<ChatRoomWorkshopForChatList>> GetByWorkshopIdsAsync(IEnumerable<Guid> workshopIds)
+        {
+            Expression<Func<ChatRoomWorkshop, bool>> condition = x => workshopIds.Contains(x.WorkshopId);
+
+            var searchMessagesForProvider = true;
```
Comment (DmyMi) [inline OutOfSchool/OutOfSchool.DataAccess/Repository/ChatRoomWorkshopModelForChatListRepository.cs:69]: maybe move this to `GetByWorkshopIdsAsync` method arguments and have true as default value?

File: OutOfSchool/OutOfSchool.IdentityServer/ProfileService.cs [hunk 2]
```diff
@@ -73,5 +81,30 @@ private async Task<string> GetPermissionsForUser(string userName, string roleNam
 
             return permissionsForUser;
         }
+
+        // Get subrole for user
+        private async Task<Subrole> GetSubroleByUserName(string userName)
+        {
+            var userToLogin = await userManager.FindByNameAsync(userName);
```
Comment (DmyMi) [inline OutOfSchool/OutOfSchool.IdentityServer/ProfileService.cs:88]: your method here and `GetPermissionsForUser` both are finding the user by name, so probaly it's worth moving it out of both methods and passing user as args.

File: OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ChatWorkshopController.cs [hunk 3]
```diff
@@ -167,7 +171,7 @@ private async Task<bool> IsProviderAChatRoomParticipantAsync(ChatRoomWorkshopDto
         {
             var userId = this.GetUserId();
 
-            var result = await validationService.UserIsWorkshopOwnerAsync(userId, chatRoom.WorkshopId).ConfigureAwait(false);
+            var result = await validationService.UserIsWorkshopOwnerAsync(userId, chatRoom.WorkshopId, GetUserSubrole()).ConfigureAwait(false);
```
Comment (DmyMi) [inline OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ChatWorkshopController.cs:174]: please don't mix passing variable and method invocation.
you already using `var userId = this.GetUserId();`, so why don't the same with `GetUserSubrole` :) especially if you're doing just that in another method

**Review by DmyMi — APPROVED** [review]

File: OutOfSchool/OutOfSchool.IdentityServer/ProfileService.cs (hunk 2 above)
Comment (Sandaljero) [reply OutOfSchool/OutOfSchool.IdentityServer/ProfileService.cs:88]: Done

File: OutOfSchool/OutOfSchool.DataAccess/Repository/ChatRoomWorkshopModelForChatListRepository.cs (hunk 1 above)
Comment (alrom) [reply OutOfSchool/OutOfSchool.DataAccess/Repository/ChatRoomWorkshopModelForChatListRepository.cs:69]: Done

File: OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ChatWorkshopController.cs (hunk 3 above)
Comment (Sandaljero) [reply OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ChatWorkshopController.cs:174]: Done

**Review by freshusername — APPROVED** [review]
LGTM

**Pull request merged**